modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/VerticalCardTag.java

```diff
@@ -0,0 +1,809 @@
+		_stickerStyle = stickerStyle;
+	}
+
+	public void setSubtitle(String subtitle) {
+		_subtitle = subtitle;
+	}
+
+	public void setTitle(String title) {
+		_title = title;
+	}
+
+	public void setVerticalCard(VerticalCard verticalCard) {
+		_verticalCard = verticalCard;
+	}
+
+	@Override
+	protected void cleanUp() {
+		super.cleanUp();
+
+		_actionDropdownItems = null;
+		_disabled = null;
+		_flushHorizontal = null;
+		_flushVertical = null;
+		_groupName = null;
+		_href = null;
+		_icon = null;
+		_imageAlt = null;
+		_imageSrc = null;
+		_inputName = null;
+		_inputValue = null;
+		_labels = null;
+		_labelStylesMap = null;
+		_selectable = null;
+		_selected = null;
+		_spritemap = null;
+		_stickerCssClass = null;
+		_stickerIcon = null;
+		_stickerImageAlt = null;
+		_stickerImageSrc = null;
+		_stickerLabel = null;
+		_stickerShape = null;
+		_stickerStyle = null;
+		_subtitle = null;
+		_title = null;
+		_verticalCard = null;
+	}
+
+	@Override
+	protected String getHydratedModuleName() {
+		return "frontend-taglib-clay/VerticalCard";
+	}
+
+	@Override
+	protected Map<String, Object> prepareProps(Map<String, Object> props) {
+		props.put("actions", getActionDropdownItems());
+		props.put("description", getSubtitle());
+		props.put("disabled", isDisabled());
+		props.put("displayType", _getDisplayType());
+		props.put("flushHorizontal", getFlushHorizontal());
+		props.put("flushVertical", getFlushVertical());
+		props.put("href", getHref());
+		props.put("id", getId());
+		props.put("imageAlt", getImageAlt());
+		props.put("imageSrc", getImageSrc());
+		props.put("inputName", getInputName());
+		props.put("inputValue", getInputValue());
+		props.put("labels", getLabels());
+		props.put("labelStylesMap", getLabelStylesMap());
+		props.put("selectable", isSelectable());
+		props.put("selected", isSelected());
+		props.put("stickerCssClass", getStickerCssClass());
+		props.put("stickerIcon", getStickerIcon());
+		props.put("stickerImageAlt", getStickerImageAlt());
+		props.put("stickerImageSrc", getStickerImageSrc());
+		props.put("stickerLabel", getStickerLabel());
+		props.put("stickerShape", getStickerShape());
+		props.put("stickerStyle", getStickerStyle());
+		props.put("symbol", getIcon());
+		props.put("title", getTitle());
+
+		return super.prepareProps(props);
+	}
+
+	@Override
+	protected String processCssClasses(Set<String> cssClasses) {
+		cssClasses.add("card-type-asset");
+
+		if (Validator.isNotNull(_imageSrc)) {
+			cssClasses.add("image-card");
+		}
+		else {
+			cssClasses.add("file-card");
+		}
+
+		if (!isSelectable()) {
+			cssClasses.add("card");
+		}
+		else {
+			cssClasses.add("form-check");
+			cssClasses.add("form-check-card");
+			cssClasses.add("form-check-top-left");
+		}
+
+		return super.processCssClasses(cssClasses);
+	}
+
+	@Override
+	protected int processStartTag() throws Exception {
+		super.processStartTag();
+
+		JspWriter jspWriter = pageContext.getOut();
+
+		if (isSelectable()) {
```

Comment:
@julien @carloslancha 

Looking at [ClayCard](https://github.com/liferay/clay/blob/master/packages/clay-card/src/Card.tsx#L68), shouldn't there be a check in root of the code  `if(getHref() == null)`, branching to `LinkTag` and `div`. Same is missing in `UserCardTag`. Is there a reason why we are skipping it?

Reply:
@markocikos what I'm seeing is that even though you pass an `href` prop to a [`CardWithInfo`](https://github.com/liferay/clay/blob/8b18dc5e48f730aa60ef1fdc51a6f75edabc3494/packages/clay-card/src/CardWithInfo.tsx) component (which renders [`Card`](https://github.com/liferay/clay/blob/8b18dc5e48f730aa60ef1fdc51a6f75edabc3494/packages/clay-card/src/Card.tsx)) the link doesn't get to be added to the root node  as can be seen in this [codesanbox](https://codesandbox.io/s/little-dawn-t8dd0).

Reply:
Oh I see, yea this is fine :+1: